### PR TITLE
Add ModelSelection view

### PR DESCRIPTION
**Pull request by jeanpimentel**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -127,6 +128,9 @@ dependencies {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
     implementation 'com.android.support:appcompat-v7:25.3.1'
+    implementation 'com.android.support:cardview-v7:25.3.1'
```
Comment (ivamluz) [inline app/build.gradle]: Better to create a variable for holding the support library version.

File: app/build.gradle [hunk 2]
```diff
@@ -127,6 +128,9 @@ dependencies {
...
+    implementation 'com.android.support:recyclerview-v7:25.3.1'
+
     implementation 'com.android.support.constraint:constraint-layout:1.0.0-beta4'
```
Comment (ivamluz) [inline app/build.gradle]: Why beta version? There is already an official version, no? (https://developer.android.com/training/constraint-layout/index.html)

File: app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            presenter.selectModel(model)
+        }
+
+        adapter.setHasStableIds(true)
+
+        recyclerView.adapter = adapter
+    }
+
+    private fun setupAboutButton() {
+        aboutButton = findViewById(R.id.about) as Button
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt]: https://github.com/JakeWharton/kotterknife?

File: app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    private lateinit var recyclerView: RecyclerView
+    private var models = ArrayList<Model>()
+
+    private lateinit var aboutButton: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_model_selection)
+
+        presenter = ModelSelectionPresenter(this)
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt]: Shouldn't we attach onResume() and detach onPause() instead? Initializing this way can cause memory leaks, no?

File: app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle?) {
+        super.onSaveInstanceState(outState)
+        outState?.putParcelableArrayList(MODELS_BUNDLE, models)
+    }
+
+    private fun setupList() {
+        recyclerView = findViewById(R.id.list) as RecyclerView
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt:60]: https://github.com/JakeWharton/kotterknife?

File: app/src/main/res/layout/activity_model_selection.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        android:layout_marginEnd="@dimen/model_selection_subtitle_horizontal_margin"
+        android:layout_below="@+id/title"
+        style="@style/ModelSelection.Subtitle"
+        android:text="@string/model_selection_subtitle"
+    />
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/list"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/model_selection_list_height"
```
Comment (ivamluz) [inline app/src/main/res/layout/activity_model_selection.xml:39]: Can't this cause accessibility problems?

File: app/src/main/res/layout/activity_model_selection.xml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          android:layout_height="@dimen/model_selection_line_height"
+          android:layout_marginLeft="@dimen/model_selection_line_horizontal_margin"
+          android:layout_marginRight="@dimen/model_selection_line_horizontal_margin"
+          android:layout_above="@+id/about"
+          android:background="@color/model_selection_line"/>
+
+    <Button
+        android:id="@+id/about"
+        android:layout_width="match_parent"
+        android:layout_height="56dp"
```
Comment (ivamluz) [inline app/src/main/res/layout/activity_model_selection.xml]: What's the criteria for putting the dimensions in-line for this component, but defining dimens for the others?

**Review by ivamluz — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt (hunk 5 above)
Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt:60]: I saw this library, but seems like an experiment, not official yet. Since we won't have many views, I found unnecessary.

File: app/src/main/res/layout/activity_model_selection.xml (hunk 6 above)
Comment (jeanpimentel) [reply app/src/main/res/layout/activity_model_selection.xml:39]: Since it is a horizontal list and the screen has no vertical scroll, I have addressed this problem to Daniel. As a PoC, he suggested a change where we remove the icon from the item if the font scale is greater than the default.

File: app/src/main/res/layout/activity_model_selection.xml (hunk 7 above)
Comment (jeanpimentel) [reply app/src/main/res/layout/activity_model_selection.xml]: A simple mistake. Fixed.

File: app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt (hunk 4 above)
Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt]: In reference projects, I see the presenter creation at onCreate method. If we have something that needs to be turned on and off then we can have the start() and stop() methods, called in onResume and onPause, but we don't need this yet.

Comment (ivamluz) [reply app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt]: I saw some reference projects proposing to presenter.attach(this) onResume() and presenter.detach() (sets view reference to null internally) onPause().

I suppose the reference to the view should be cleaned, otherwise we may run into memory leaks on configuration changes, for example, no?

File: app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt (hunk 5 above)
Comment (ivamluz) [reply app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt:60]: Ok.

File: app/src/main/res/layout/activity_model_selection.xml (hunk 6 above)
Comment (ivamluz) [reply app/src/main/res/layout/activity_model_selection.xml:39]: Ok.

File: app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt (hunk 4 above)
Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt]: Changed!

**Review by ivamluz — APPROVED** [review]
